### PR TITLE
Fix missing concurrent scan finish notification

### PR DESCRIPTION
**Pull request by LinHu2016**
As part of synchronization between concurrent scanning and continuation mounting, the continuation mounting could be blocked by concurrent scanning, release vmaccess and waiting for current thread mutex, but there is timing hole between release vmaccess and enter the mutex, the notification of the mutex(the end of concurrent scanning) could be sent before the enter of current thread mutex, then cause the mounting thread would wait forever(hang).

Enter the mutex earlier in order to avoid the time hole.

fix: https://github.com/eclipse-openj9/openj9/issues/17163

### CODE REVIEW

File: runtime/oti/VMHelpers.hpp [hunk 1]
```diff
@@ -2241,21 +2241,28 @@ class VM_VMHelpers
 		/* clear CONCURRENTSCANNING flag bit3:LocalConcurrentScanning /bit4:GlobalConcurrentScanning */
 		ContinuationState oldContinuationState = 0;
 		ContinuationState returnContinuationState = 0;
+		J9VMThread *carrierThread = NULL;
+		bool enteredMonitor = false;
 		do {
 			oldContinuationState = *continuationStatePtr;
+			if (!enteredMonitor) {
```
Comment (amicic) [inline runtime/oti/VMHelpers.hpp:2248]: might be an overkill, but if enteredMonitor is true we could assert that  getCarrierThread(oldContinuationState) is still carrierThread

File: runtime/vm/ContinuationHelpers.cpp [hunk 2]
```diff
@@ -110,25 +110,28 @@ synchronizeWithConcurrentGCScan(J9VMThread *currentThread, j9object_t continuati
...
-
-			internalAcquireVMAccess(currentThread);
-			continuationObject = POP_OBJECT_IN_SPECIAL_FRAME(currentThread);
-			/* Object might have moved - update its address and the address of the state slot. */
-			continuationStatePtr = VM_VMHelpers::getContinuationStateAddress(currentThread, continuationObject);
-			returnContinuationState = *continuationStatePtr;
+			/* Wait till potentially concurrent GC scans are complete */
+			do {
+				oldContinuationState = *continuationStatePtr;
+			    if (VM_VMHelpers::isConcurrentlyScanned(oldContinuationState)) {
```
Comment (amicic) [inline runtime/vm/ContinuationHelpers.cpp:118]: the whole 'if' block should be indented

File: runtime/oti/VMHelpers.hpp (hunk 1 above)
Comment (LinHu2016) [reply runtime/oti/VMHelpers.hpp:2248]: we can add assert after move exitConcurrentGCScan() to GCExtensions.cpp

Comment (amicic) [reply runtime/oti/VMHelpers.hpp:2248]: ah, we can't assert here because this is *hpp? ok, just don't do it (not even sure that assert after is correct)!?

**Comment by LinHu2016** [comment]
tested with a random sleep before omrthread_monitor_enter() in synchronizeWithConcurrentGCScan().

**Comment by amicic** [comment]
@gacholio and @babsingh
please, have a look

**Review by gacholio — APPROVED** [review]

**Comment by amicic** [comment]
jenkins test sanity aix,win jdk19

**Review by babsingh — APPROVED** [review]

**Review by amicic — APPROVED** [review]

**Pull request merged**